document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java

```diff
@@ -124,141 +124,22 @@ public void testCollections() {
 
   @Test
   public void testIgnoreCaseLikeQuery() throws IOException {
-    long now = Instant.now().toEpochMilli();
-    Collection collection = datastore.getCollection(COLLECTION_NAME);
-    collection.upsert(
-        new SingleValueKey("default", "testKey"), Utils.createDocument("name", "Bob"));
-
-    String[] ignoreCaseSearchValues = {"Bob", "bob", "BOB", "bOB", "BO", "bO", "Ob", "OB"};
-
-    for (String searchValue : ignoreCaseSearchValues) {
-      Query query = new Query();
-      query.setFilter(new Filter(Filter.Op.LIKE, "name", searchValue));
-      Iterator<Document> results = collection.search(query);
-      List<Document> documents = new ArrayList<>();
-      while (results.hasNext()) {
-        documents.add(results.next());
-      }
-      Assertions.assertFalse(documents.isEmpty());
-      String persistedDocument = documents.get(0).toJson();
-      JsonNode jsonNode = OBJECT_MAPPER.reader().readTree(persistedDocument);
-      Assertions.assertTrue(persistedDocument.contains("Bob"));
-      Assertions.assertTrue(jsonNode.findValue("createdTime").asLong(0) >= now);
-      Assertions.assertTrue(jsonNode.findValue("lastUpdatedTime").asLong(0) >= now);
-    }
+    DocStoreTest.testIgnoreCaseLikeQuery(datastore, DocStoreTest.MONGO_STORE);
```

Comment:
- Can we mark this common test as `@Test` in DocStoreTest?
- And in the DocStoreTest.testIgnoreCaseLikeQuery, can we have for loop for each data-store?

So, this way, all the common tests will be moved to DocStoreTest.

Reply:
Will do it in next pr.

Reply:
ok, let's see if possible to parametrized the test with document-store input.